File: python-wrapper/src/python/model/embed/Gender.py
from typing import Union, Dict
from enum import Enum

class Gender(Enum):
	Male = "male"
	Female = "female"
	Indeterminate = "indeterminate"
	Changed = "changed"
	ChangedToMale = "changedToMale"
	ChangedToFemale = "changedToFemale"
	Unknown = "unknown"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'Gender':
		if data == "male":
			return Gender.Male
		elif data == "female":
			return Gender.Female
		elif data == "indeterminate":
			return Gender.Indeterminate
		elif data == "changed":
			return Gender.Changed
		elif data == "changedToMale":
			return Gender.ChangedToMale
		elif data == "changedToFemale":
			return Gender.ChangedToFemale
		elif data == "unknown":
			return Gender.Unknown
		else:
			raise Exception(f"{data} is not a valid value for Gender enum.")

__all__ = ['Gender']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/Gender.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class Gender(
	public val code: String,
	internal val dtoSerialName: String,
) {
	@SerialName("male")
	Male("M", "male"),

	@SerialName("female")
	Female("F", "female"),

	@SerialName("indeterminate")
	Indeterminate("I", "indeterminate"),

	@SerialName("changed")
	Changed("C", "changed"),

	@SerialName("changedToMale")
	ChangedToMale("Y", "changedToMale"),

	@SerialName("changedToFemale")
	ChangedToFemale("X", "changedToFemale"),

	@SerialName("unknown")
	Unknown("U", "unknown"),
}
